test/boost/virtual_table_test.cc

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2021 ScyllaDB
+ *
+ * Modified by ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <seastar/testing/test_case.hh>
+#include "test/lib/test_services.hh"
+
+#include "db/virtual_table.hh"
+#include "db/system_keyspace.hh"
+
+namespace db {
+
+class test_table : public virtual_table {
+public:
+    test_table() : virtual_table(build_schema()) {}
+
+    static schema_ptr build_schema() {
+        auto id = generate_legacy_id(system_keyspace::NAME, "test");
+        return schema_builder(system_keyspace::NAME, "test", std::make_optional(id))
+            .with_column("pk", int32_type, column_kind::partition_key)
+            .with_column("ck", int32_type, column_kind::clustering_key)
+            .with_column("v", int32_type)
+            .with_version(system_keyspace::generate_schema_version(id))
+            .build();
+    }
+
+    mutation_source as_mutation_source() override {
+        throw std::runtime_error("Not implemented");
+    }
+
+    void test_set_cell() {
+        mutation m(_s, partition_key::from_single_value(*_s, data_value(666).serialize_nonnull()));
+        row& cr = m.partition().clustered_row(*_s, clustering_key::from_single_value(*_s, data_value(10).serialize_nonnull())).cells();
+
+        set_cell(cr, "v", 8);
+
+        auto result_cell = cr.cell_at(0).as_atomic_cell(column_definition("v", int32_type, column_kind::regular_column));
+        auto result = result_cell.serialize();
+
+        BOOST_REQUIRE(result[result.size() - 1] == 8);
+        BOOST_REQUIRE(abs(result_cell.timestamp() - api::new_timestamp()) < 100);
```

Comment:
A stall in test execution will fail this assert. It's bad to rely on timing. Maybe don't test it, or parametrize set_cell() with timestamp value.

Reply:
I put a difference there that I thought should be big enough for common stalls
But I agree that this case is not that important in this context. I'll just remove it